### DISCUSSION TITLE
Not getting good realistic results with Hyper-SD + IP-Adapter

### DISCUSSION THREAD

**Discussion by juancopi81** [huggingface/diffusers#7818] [General]
Hi everyone, (maybe you @asomoza know about this?)

Does hyper-sd works well with IP-Adapter? I am testing hyper-sd in Diffusers as explained in the [repo](https://huggingface.co/ByteDance/Hyper-SD). I thought that I was going to get better results than LCM-LoRA but my results so far are in fact bad when used with IP-Adapter. They seem very cartoonish to me.

Here is the code I am using, taken from the Model Card in HF:

```py
!pip install -U diffusers peft
!pip list | grep diffusers # -> 0.27.2

import torch
from diffusers.utils import load_image
import numpy as np
import cv2
from PIL import Image
# Tests with different schedulers
from diffusers import ControlNetModel, StableDiffusionControlNetPipeline, DDIMScheduler, TCDScheduler, LCMScheduler
from huggingface_hub import hf_hub_download

controlnet_checkpoint = "lllyasviel/control_v11p_sd15_openpose"
# Load original image
cn_image = load_image("assets/neutral_openpose_woman.jpg")

# Initialize pipeline
controlnet = ControlNetModel.from_pretrained(controlnet_checkpoint, torch_dtype=torch.float16)
pipe = StableDiffusionControlNetPipeline.from_pretrained("SG161222/Realistic_Vision_V6.0_B1_noVAE", controlnet=controlnet, torch_dtype=torch.float16).to("cuda")
pipe.load_lora_weights(hf_hub_download("ByteDance/Hyper-SD", "Hyper-SD15-8steps-lora.safetensors"))
pipe.fuse_lora()

face_image = load_image("https://huggingface.co/datasets/YiYiXu/testing-images/resolve/main/women_input.png")
pipe.load_ip_adapter("h94/IP-Adapter", subfolder="models", weight_name="ip-adapter-full-face_sd15.bin")
pipe.set_ip_adapter_scale(0.85)

# Ensure ddim scheduler timestep spacing set as trailing !!!
# Tested with different schedulers: DDIM, LCM, TCD...
# pipe.scheduler = DDIMScheduler.from_config(pipe.scheduler.config, timestep_spacing="trailing")
pipe.scheduler = TCDScheduler.from_config(pipe.scheduler.config)
prompt="RAW photo, a photo of a woman wearing jeans and a t-shirt. Best quality. hyper-realistic photo, detailed face"
negative_prompt="worst quality, low quality, extra fingers, extra arms, extra ears"
image = pipe(prompt, negative_prompt=negative_prompt, num_inference_steps=8, image=cn_image, guidance_scale=0, eta=1.0, ip_adapter_image=face_image).images[0]
image.save('image_out.png')
```

These are the relevante images:
**ControlNet Image:**
![neutral_openpose_woman](https://github.com/huggingface/diffusers/assets/4313860/17a62c00-3e66-40b7-b774-284cda6703b5)

**IP-Adaper Image:**
![face_image](https://github.com/huggingface/diffusers/assets/4313860/2f524ed9-9eec-43b5-a33b-42114d5601c0)

**Resulting Image:**
![failed_hyper_sd](https://github.com/huggingface/diffusers/assets/4313860/439a1041-127a-4a66-9a82-22574c64a11e)

For comparison, if I keep everything the same and use LCM-LoRA, changing only these lines:

```py
pipe.load_lora_weights("latent-consistency/lcm-lora-sdv1-5")
pipe.scheduler = LCMScheduler.from_config(pipe.scheduler.config)
```
I get this much more realistic image:
![lcm_lora](https://github.com/huggingface/diffusers/assets/4313860/e9321c89-4690-493a-b100-5832049d1652)

Not sure if I am doing something wrong or if it is just that ip-adapter does not yield realistic results with hyper-sd. Any help will be highly appreciated!

Thanks!

**Comment by asomoza** [comment]
I haven't really tested Hyper-SD with SD 1.5, when I have time later I can do it. But with SDXL I don't have any problems, just a quick test since I can't download all what you're using right now, but with a depth controlnet and your images.

Using RealVis, TCD and the 8-step LoRA:

|without mask|with mask|
|---|---|
|![20240429140759](https://github.com/huggingface/diffusers/assets/5442875/21726be5-dcaf-42d2-ab07-f8490cf8eaff)|![20240429140819](https://github.com/huggingface/diffusers/assets/5442875/6b0ef6f6-636b-476c-9b4f-279b8e60b9bd)|

If I just use the IP Adapter 

|normal|plus|
|---|---|
|![20240429140921](https://github.com/huggingface/diffusers/assets/5442875/30f4acb5-8cc2-46d6-8b56-5741cfa564ad)|![20240429140933](https://github.com/huggingface/diffusers/assets/5442875/27b7809e-d832-4858-9fc1-b09a378dfd8d)|

So for now I can reduce that what's happening with you is either the controlnet (guidance to high?) or just that SD 1.5 can't handle  it.

**Reply by juancopi81** [reply]
Thank you very much for your answer @asomoza, I was playing with different conditioning scales, but still very cartoonish, maybe it is how it behaves for SD 1.5 🤔

**Reply by asomoza** [reply]
I did some tests with this, I'm not that familiar with SD 1.5 anymore and also I'm too used to the SDXL results, so I can't really tell when something is good or bad with SD 1.5 unless it's really bad.

With your IP Face Image, If I use a Plus Face IP Adapter:

|Euler|DPMSolverMultistep|
|---|---|
|![20240508222637](https://github.com/huggingface/diffusers/assets/5442875/4b6fcc38-ff9c-4712-976c-32a90692b092)|![20240508222828](https://github.com/huggingface/diffusers/assets/5442875/8682cca9-2f79-456c-997a-211a1d65210b)|

And with Hyper-SD:

|TCD|DPMSolverMultistep|
|---|---|
|![20240508223716](https://github.com/huggingface/diffusers/assets/5442875/9dd888a3-12fa-44ef-a74d-e58e754dd09e)|![20240508224120](https://github.com/huggingface/diffusers/assets/5442875/7a3cd755-b0b0-4a7c-b0d2-fac545cda30f)|

All the results don't look real to me, but the Hyper-SD results look more cartoonish. Still they're not as bad as the image you posted.

Bytedance released a new one that can be used with CFG: `Hyper-SD15-8steps-CFG-lora.safetensors`, with that one and a CFG of 1.8, it looks decent, and with 3.0 it looks like the old one. 

I think you're right, it has to do with how SD 1.5 behaves. The normal Hyper-SD with IP Adapters behaves like if it has a high CFG which makes the results over saturated and cartoonish.

|CFG 2.0|CFG 3.0|
|---|---|
|![20240508224510](https://github.com/huggingface/diffusers/assets/5442875/c16939d5-4396-43e8-8f6b-378405466d71)|![20240508224628](https://github.com/huggingface/diffusers/assets/5442875/d522342b-8a20-4fcf-b532-e69f1fb23c55)|

**Reply by juancopi81** [reply]
Thanks @asomoza! I'll give it a try to `Hyper-SD15-8steps-CFG-lora.safetensors`! Hopefully works better with the ip-adapter-

**Reply by juancopi81** [reply]
Yeap! I can confirm this one is working much better

**Reply by juancopi81** [reply]
Maybe the ip-adapter needs some CFG in SD1.5 so that's why this model works better 🤔
